### DISCUSSION TITLE
Brier Score Function Parameters

### DISCUSSION THREAD

**Discussion by phaniparsa** [sebp/scikit-survival#173] [Q&A]
Hello All,

As per https://scikit-survival.readthedocs.io/en/latest/api/generated/sksurv.metrics.brier_score.html , brier score method accepts four parameters, which are:
1. survival_train: survival times for training data to estimate censoring distribution from.
2. survival_test: survival times of the test data.
3. estimate: survival probabilities given by a model (say RSF).
4. times: The time points for which to estimate the Brier score. Values must be within the range of follow-up times of the test data survival_test

Upon exploring RSF predict_survival output, survival probabilities needs to be forward filled (where as in COX PH it appears that output returned is a step function which automatically takes care of forward filling part) and all the times between min(survival test times) and max(survival test times)-1. Will anyone of you help me whether this understanding is correct?

Also, in case if the dataset for survival analysis does not contain censored cases, will it be ok to fall back to the previous version of brier score (I understand that the current brier score is adjusted to reflect censored cases in '98)? 

Thanks,
Sai Phani Parsa.

**Comment by sebp** [comment]
If you call RSF's `predict_survival_function` with `return_array=False`, you get the same output as for CoxPH, an array of step functions.

Yes, this version of the Brier score does re-weighting based on the probability of being censored. If you're data as no censored times, then the regular Brier score is fine, but then again you wouldn't need any of the models in scikit-survival to begin with.

**Reply by phaniparsa** [reply]
Thank you for the reply. 

1. We have a number of independent variables in our case study and trying to capture any non-linear relationships between the target variable, and predict the survival probabilities at a given time t. Any suggestions on whether we can use RSF from scikit-survival or not?

2. For the current scikit-surv Brier Score and IBS Score implementation, upon reviewing the code, I'm of the understanding that "times" parameter should be with in range of min(test set times) and one less than max(test set times). Am I correct?

Thanks,
Sai Phani Parsa

**Comment by sebp** [comment]
1. You can still use a survival model, there's nothing wrong with that, 
in particular if you want to estimate survival functions. As you 
mentioned, using standard regression would be only valid in the 
non-censored case.

2. Yes, you are correct.
